### PR TITLE
[GHF] Fix ghstack branches in sync logic

### PR DESCRIPTION
**Pull request by malfet**
Test plan:
```python
from git_utils import are_ghstack_branches_in_sync,GitRepo
repo=GitRepo("/Users/nshulga/git/pytorch/pytorch")
are_ghstack_branches_in_sync(repo, "gh/SS-JIA/206/head")
```

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -619,6 +620,7 @@ def can_skip_internal_checks(pr: "GitHubPR", comment_id: Optional[int] = None) -
         return False
     return comment.author_login == "facebook-github-bot"
 
+
```
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:623]: nit: extra newline

**Review by ZainRizvi — APPROVED** [review]

File: .github/scripts/trymerge.py (hunk 1 above)
Comment (malfet) [reply .github/scripts/trymerge.py:623]: Err, isn't this a convention? Two blank lines between top level functions?
https://peps.python.org/pep-0008/#blank-lines

Comment (ZainRizvi) [reply .github/scripts/trymerge.py:623]: I suppose it's an official convention, but I haven't noticed it being done to often in our repo.

Looking at this file, maybe a bit under half of the top level functions are separated this way...eh, whatevs. This is where lint-consistently-on-save tools are handy :P

**Pull request closed without merging**